src/main/java/com/juniorstart/juniorstart/service/UserTechnologyService.java

```diff
@@ -0,0 +1,68 @@
+package com.juniorstart.juniorstart.service;
+
+
+import com.juniorstart.juniorstart.exception.ResourceNotFoundException;
+import com.juniorstart.juniorstart.model.UserProfile;
+import com.juniorstart.juniorstart.model.UserTechnology;
+import com.juniorstart.juniorstart.repository.UserProfileRepository;
+import com.juniorstart.juniorstart.repository.UserTechnologyRepository;
+import com.juniorstart.juniorstart.security.UserPrincipal;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import java.util.Optional;
+import java.util.Set;
+
+@Service
+@Slf4j
+public class UserTechnologyService {
+
+    final private UserProfileRepository userProfileRepository;
+
+    final private UserTechnologyRepository userTechnologyRepository;
+
+    public UserTechnologyService(UserProfileRepository userProfileRepository, UserTechnologyRepository userTechnologyRepository){
+        this.userProfileRepository = userProfileRepository;
+        this.userTechnologyRepository = userTechnologyRepository;
+    }
+
+    public ResponseEntity<?> addUserTechnology(Set<UserTechnology> userTechnology, UserPrincipal currentUser){
+
+        Optional<UserProfile> foundUser = Optional.ofNullable(userProfileRepository.findByPrivateId(currentUser.getId()).orElseThrow(() ->
+                new ResourceNotFoundException("UserProfile", "ID", currentUser.getId())));
+
```

Comment:
Tego używasz 3 razy w różnych metodach. Wyciągnij to do nowej metody i odwołaj się gdzie trzeba.

Reply:
Faktycznie nie zauważyłem, poprawione